src/util/strencodings.cpp

```diff
@@ -508,17 +509,36 @@ std::string Capitalize(std::string str)
     return str;
 }
 
+namespace {
+
+using ByteAsHex = std::array<char, 2>;
+
+constexpr std::array<ByteAsHex, 256> CreateByteToHexMap()
+{
+    std::array<ByteAsHex, 256> byte_to_hex{};
+    for (size_t i = 0; i < byte_to_hex.size(); ++i) {
+        char const* hex = "0123456789abcdef";
+        byte_to_hex[i][0] = hex[i >> 4];
+        byte_to_hex[i][1] = hex[i & 15];
+    }
+    return byte_to_hex;
+}
+
+} // namespace
+
 std::string HexStr(const Span<const uint8_t> s)
 {
     std::string rv(s.size() * 2, '\0');
-    static constexpr char hexmap[16] = { '0', '1', '2', '3', '4', '5', '6', '7',
-                                         '8', '9', 'a', 'b', 'c', 'd', 'e', 'f' };
-    auto it = rv.begin();
+    static constexpr auto byte_to_hex = CreateByteToHexMap();
+    static_assert(sizeof(byte_to_hex) == 512);
+
+    char* it = rv.data();
     for (uint8_t v : s) {
-        *it++ = hexmap[v >> 4];
-        *it++ = hexmap[v & 15];
+        std::memcpy(it, byte_to_hex[v].data(), 2);
+        it += 2;
```

Comment:
```suggestion
        const auto& h{byte_to_hex[v]};
        it = std::copy(h.begin(),h.end(), it);
```

Side-note: Looks like it requires gcc-12 to get the same binary if this is written with `std::copy`.

Reply:
I also find it interesting that clang unrolls the loop, but gcc doesn't. I tried to force unrollment with `#pragma GCC unroll 8` and that does the job, but still the code that comes out is not faster in my benchmark. 
https://godbolt.org/z/81x66rP1c (I've replaced `Span` with `string_view` but it should be practically the same)